trieste/acquisition/function.py

```diff
@@ -405,3 +410,93 @@ def probability_of_feasibility(
     mean, var = model.predict(at)
     distr = tfp.distributions.Normal(mean, tf.sqrt(var))
     return distr.cdf(tf.cast(threshold, at.dtype))
+
+
+class MonteCarloAcquisition(SingleModelAcquisitionBuilder):
+    """
+    Builder for the expected improvement function where the "best" value is taken to be the minimum
+    of the posterior mean at observed points.
+    """
+
+    def __init__(self, eps_shape: [int]):
+
+        super().__init__()
+        assert len(eps_shape) == 3
+        self.eps = tf.random.normal(eps_shape, dtype=default_float())  # [S, B, L]
```

Comment:
for simplicity now the user specifies everything. Actually he should only have to care about S; B is one unless you choose a batch-sequential strategy but this should be given by the rule, and L is the number of latent processes that can be inferred from the model.

Reply:
do we support batch sequential? cos if not, i suggest we don't make it an option

Reply:
why should batch size be determined by the rule instead of the acquisition function?

Reply:
We will support batches (cf https://github.com/secondmind-labs/trieste/issues/41). 

I think that the rule is the proper place to define the batch size.